controllers/machineset.go

```diff
@@ -109,11 +114,12 @@ func scaleMachineSet(aw *arbv1.AppWrapper, userRequestedInstanceType string, rep
 					//TODO: user can delete appwrapper work on triggering scale-down
 					klog.Infof("waiting for machines to be in state Ready. replicas needed: %v and replicas available: %v", replicas, ms.Status.AvailableReplicas)
 					time.Sleep(1 * time.Minute)
```

Comment:
The reconcile loop should not be blocked. Instead the reconcile request should be re-queued, so the state of the MachineSet is checked.